test/feature-benchmark/mzcompose.py

```diff
@@ -132,29 +136,14 @@ def run_one_scenario(
                 param_name, param_value = param.split("=")
                 additional_system_parameter_defaults[param_name] = param_value
 
-        mz = Materialized(
-            image=f"materialize/materialized:{tag}" if tag else None,
-            default_size=size,
-            # Avoid clashes with the Kafka sink progress topic across restarts
-            environment_id=f"local-az1-{uuid.uuid4()}-0",
-            soft_assertions=False,
-            additional_system_parameter_defaults=additional_system_parameter_defaults,
-            external_cockroach=True,
-            external_minio=True,
-            sanity_restart=False,
-        )
+        mz_image = f"materialize/materialized:{tag}" if tag else None
 
-        with c.override(mz):
-            print(f"The version of the '{instance.upper()}' Mz instance is:")
-            c.run(
-                "materialized",
-                "-c",
-                "environmentd --version | grep environmentd",
-                entrypoint="bash",
-                rm=True,
-            )
+        if mz_image is not None and not c.try_pull_single_image(mz_image):
```

Comment:
`pull_single_image` takes the service name as a argument, not the image name, as `docker compose pull` takes a service name:

```
Usage:  docker compose pull [OPTIONS] [SERVICE...]

Pull service images
```

So there are two options here:

1. Call `pull_single_image` with `materialized` as an argument,  but in an `override` block that sets `Materialized(image...)`
2. call `docker pull` directly rather than `docker-compose pull`. There must already be a method in misc/python/materialize that uses `docker pull` and not `docker compose pull`

Reply:
Thank you!

✅ Fixed with https://github.com/MaterializeInc/materialize/pull/22355/commits/48d180e5648ba6c11e77d059d4cabb9cf604a867 and successfully tested locally.